raiden/network/transport/matrix.py

```diff
@@ -418,7 +431,7 @@ def _post_sync(self, next_batch):
         so that the token is used as a starting point from which
         messages are fetched from the matrix server.
         """
-        state_change = ActionUpdateTransportSyncToken(next_batch)
+        state_change = ActionUpdateTransportSyncToken(f'{self._server_name}/{next_batch}')
```

Comment:
This is basically invalidating the assumption that all servers should be synced so that the token will work on any of them. However, we actually want that assumption to be true... whats the reasoning here?

Reply:
Users are tied to a homeserver and the sync token is tied to a specific to a user. If we're connecting to a different server we automatically use a different matrix user and therefore can't re-use the sync token.

Reply:
hmm that makes sense. But what happens when i shutdown my node and re-launch it and Raiden automatically uses another server... is there any potential loss of messages due to this not syncing properly?

Reply:
You could add use `self._user_id` instead, to be triple sure

Reply:
@andrevmatos Yes, good idea.

@rakanalh When a new user for a known address comes online they will automatically be picked up and the peers will invite them into the appropriate rooms (or create new ones if necessary). From that point on the retry will work as normal and make sure any messages that are possibly lost on another server will be re-sent.